### PR TITLE
Day05 additions

### PR DESCRIPTION
**Pull request by kghite**
Scrapped the airplane thing, because looking over it, it didn't really move towards a learning goal other than making sure that their hashmap worked.  Instead, I added two challenge problems that should only take them a few minutes to code, pair with problems that I put in the in class pset docs, and make them think about how the runtime compares with previous map/list structures.

### CODE REVIEW

**Comment by davidabrahams** [comment]
Why is a Hashmap necessary for the solutions (as opposed to a Set)? They just need to keep track of which integers they've seen, right?

File: day05/src/HashMapProblemsTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class HashMapProblemsTest {
+
+    /**
+     * Test method for int countUnique(int[] arr)
+     */
+    @Test
+    public void testCountUnique() {
+        int[] arr = {0, 5, 6, 5, 9, 1, 6, 5, 3};
+
+        assertThat(soln.HashMapProblems.countUnique(arr), is(6));
```
Comment (thecardkid) [inline day05/src/HashMapProblemsTest.java]: There needs to be more test cases, like:
[1,1,1,1,1] --> 1
[] --> 0
[1,2,3,4,5] --> 5

File: day05/src/HashMapProblems.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+public class HashMapProblems {
+
+    public static int countUnique(int[] arr) {
+        // Return the number of unique elements in the input array
+
+        // TODO: Your code here
+    }
+
+    public static List<Integer> removeDuplicates(int[] arr) {
```
Comment (thecardkid) [inline day05/src/HashMapProblems.java]: Can you come up with a problem more different than the other one? This one would just be calling `countUnique` and returning `map.keySet()` as a list

File: day05/README.md [hunk 3]
```diff
@@ -31,4 +31,6 @@ Briefly describe the key differences and improvements between MyLinearMap and My
 
 ## Problems
 
-Fill in the missing functions in AirTrafficControl.java following the instructions in the file.
+* Complete the function `countUnique(int[] arr)`, which should take in an unsorted list of integers and return the number of unique integers in the list.  For example, given the list `[0, 5, 6, 5, 9, 1, 6, 5, 3]`, the function should return `6`.  Describe the runtime complexity of your solution in a comment as well as how your solution would differ if you did not have the HashMap data structure.
```
Comment (thecardkid) [inline day05/README.md]: Looking at the solution, these are really not intermediate problems - they were really too easy. Not saying that I prefer impossible problems, but problems not as immediately straightforward

File: day05/src/HashMapProblems.java (hunk 2 above)
Comment (thecardkid) [inline day05/src/HashMapProblems.java]: A really problem to ask here would be the thing they will have to do in the search engine: convert data that looks like this
```
{
"John": {
  "CD": 4,
  "Book": 2
},
"Sarah": {
  "CD": 3,
  "Pen": 1
}
}
```
to data that looks like this:
```
{
"Book": [ John ],
"CD": [ John, Sarah ],
"Pen": [ Sarah ]
}
```

Comment (thecardkid) [inline day05/src/HashMapProblems.java]: Something like you're giving a map of people's possessions and count, convert that to a map from item to people who own them for better search time by item.

**Pull request closed without merging**